pkg/apis/openstack/validation/infrastructure.go

```diff
@@ -83,7 +83,12 @@ func ValidateInfrastructureConfig(infra *api.InfrastructureConfig, nodesCIDR *st
 func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *api.InfrastructureConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.Networks, oldConfig.Networks, fldPath.Child("networks"))...)
+	newNetworks := newConfig.Networks
+	oldNetworks := oldConfig.Networks
+	// share network changes are allowed, therefore ignore them on comparing
+	newNetworks.ShareNetwork = nil
+	oldNetworks.ShareNetwork = nil
```

Comment:
Is there any cause for concern for disabling a `shareNetwork` while there are volumes created ?

Reply:
In this case the deletion of the `shareNetwork` will fail on the Openstack side. As a user you can fix it by removing the PVC/PV as long as the CSI manila driver is enabled.